parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -298,8 +299,14 @@ public EncodingStats convertEncodingStats(List<PageEncodingStats> stats) {
     return formatStats;
   }
 
+  @Deprecated
   public static Statistics toParquetStatistics(
       org.apache.parquet.column.statistics.Statistics statistics) {
+    return toParquetStatistics(statistics, null);
+  }
+
+  public static Statistics toParquetStatistics(
+      org.apache.parquet.column.statistics.Statistics statistics, PrimitiveType type) {
```

Comment:
Statistics should keep a reference to the type it was created with and use it here. That would make a lot of the changes to pass `type` unnecessary.

Reply:
Agreed. Will do it.